etc/docker/init-vnext-core.sh

```
@@ -0,0 +1,616 @@
+API_ENDPOINT="${VNEXT_APP_URL}/api/v1/admin/publish"
+
+# Get APP_DOMAIN from environment variable, default to "core" if not set
+APP_DOMAIN=${APP_DOMAIN:-"core"}
+echo "Using APP_DOMAIN: $APP_DOMAIN"
+
+# Wait for the vnext-app to be ready with health check
+# Wait for the vnext-app to be ready with health check (with max retries)
+MAX_RETRIES=24  # Wait up to 2 minutes (24 * 5 seconds)
+RETRY_COUNT=0
+while true; do
+    if curl -s "${VNEXT_APP_URL}/health" > /dev/null 2>&1; then
```

Shellcheck (0.11.0)

[error] Couldn't find 'fi' for this 'if'.

(SC1046)